src/components/Table/TableHead/ColumnHeaderRow/ColumnHeaderRow.test.jsx

```diff
@@ -52,3 +53,63 @@ describe('TableHead', () => {
     expect(commonTableHeadProps.onChangeOrdering).toHaveBeenCalled();
   });
 });
+
+describe('ColumnHeaderRow test', () => {
+  test('when hasRowExpansion set to true', () => {
+    const tableHeadProps = {
+      /** List of columns */
+      columns: [
+        { id: 'col1', name: 'Column 1', isSortable: false },
+        { id: 'col2', name: 'Column 2', isSortable: false },
+      ] /** Ordering list */,
+      ordering: [{ columnId: 'col1', isHidden: false }, { columnId: 'col2', isHidden: false }],
+      tableOptions: {
+        hasRowSelection: 'multi',
+        hasRowExpansion: true,
+      },
+      onChangeOrdering: jest.fn(),
+    };
+
+    const renderedElement = render(<UnconnectedColumnHeaderRow {...tableHeadProps} />);
+    expect(renderedElement.container.innerHTML).toContain('colspan="2"');
```

Comment:
instead of checking the colspan, wouldn't it be better to verify that the right number of "extra" column headers rendered in each case?